Subject: [PATCH] 3_09_add_junit
---
Index: pom.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/pom.xml b/pom.xml
--- a/pom.xml	(revision 3b4b9f1d0bbe1b4cac5e6706073ca3c59a532631)
+++ b/pom.xml	(date 1729116009503)
@@ -20,6 +20,9 @@
         <!-- Logging -->
         <logback.version>1.2.13</logback.version>
         <slf4j.version>1.7.36</slf4j.version>
+
+        <!--    Tests  -->
+        <junit.version>4.13.2</junit.version>
     </properties>
 
     <build>
@@ -40,6 +43,14 @@
                     <target>${java.version}</target>
                 </configuration>
             </plugin>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-surefire-plugin</artifactId>
+                <version>2.22.2</version>
+                <configuration>
+                    <argLine>-Dfile.encoding=UTF-8</argLine>
+                </configuration>
+            </plugin>
         </plugins>
     </build>
 
@@ -84,6 +95,14 @@
             <artifactId>jstl</artifactId>
             <version>1.2</version>
         </dependency>
+
+        <!--Test-->
+        <dependency>
+            <groupId>junit</groupId>
+            <artifactId>junit</artifactId>
+            <version>${junit.version}</version>
+            <scope>test</scope>
+        </dependency>
     </dependencies>
 
     <profiles>
Index: src/test/java/ru/javawebinar/topjava/UserTestData.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/test/java/ru/javawebinar/topjava/UserTestData.java b/src/test/java/ru/javawebinar/topjava/UserTestData.java
new file mode 100644
--- /dev/null	(date 1729116009310)
+++ b/src/test/java/ru/javawebinar/topjava/UserTestData.java	(date 1729116009310)
@@ -0,0 +1,15 @@
+package ru.javawebinar.topjava;
+
+import ru.javawebinar.topjava.model.Role;
+import ru.javawebinar.topjava.model.User;
+
+public class UserTestData {
+    public static final int USER_ID = 1;
+    public static final int ADMIN_ID = 2;
+    public static final int GUEST_ID = 3;
+    public static final int NOT_FOUND = 10;
+
+    public static final User user = new User(USER_ID, "User", "user@yandex.ru", "password", Role.USER);
+    public static final User admin = new User(ADMIN_ID, "Admin", "admin@gmail.com", "admin", Role.ADMIN);
+    public static final User guest = new User(GUEST_ID, "Guest", "guest@gmail.com", "guest");
+}
Index: src/main/java/ru/javawebinar/topjava/repository/inmemory/InMemoryBaseRepository.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/java/ru/javawebinar/topjava/repository/inmemory/InMemoryBaseRepository.java b/src/test/java/ru/javawebinar/topjava/repository/inmemory/InMemoryBaseRepository.java
rename from src/main/java/ru/javawebinar/topjava/repository/inmemory/InMemoryBaseRepository.java
rename to src/test/java/ru/javawebinar/topjava/repository/inmemory/InMemoryBaseRepository.java
--- a/src/main/java/ru/javawebinar/topjava/repository/inmemory/InMemoryBaseRepository.java	(revision 3b4b9f1d0bbe1b4cac5e6706073ca3c59a532631)
+++ b/src/test/java/ru/javawebinar/topjava/repository/inmemory/InMemoryBaseRepository.java	(date 1729116009316)
@@ -9,9 +9,9 @@
 
 public class InMemoryBaseRepository<T extends AbstractBaseEntity> {
 
-    private static final AtomicInteger counter = new AtomicInteger(0);
+    static final AtomicInteger counter = new AtomicInteger(0);
 
-    private final Map<Integer, T> map = new ConcurrentHashMap<>();
+    final Map<Integer, T> map = new ConcurrentHashMap<>();
 
     public T save(T entity) {
         if (entity.isNew()) {
@@ -33,4 +33,8 @@
     Collection<T> getCollection() {
         return map.values();
     }
+
+    void put(T entity) {
+        map.put(entity.getId(), entity);
+    }
 }
\ No newline at end of file
Index: src/main/java/ru/javawebinar/topjava/repository/inmemory/InMemoryMealRepository.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/java/ru/javawebinar/topjava/repository/inmemory/InMemoryMealRepository.java b/src/test/java/ru/javawebinar/topjava/repository/inmemory/InMemoryMealRepository.java
rename from src/main/java/ru/javawebinar/topjava/repository/inmemory/InMemoryMealRepository.java
rename to src/test/java/ru/javawebinar/topjava/repository/inmemory/InMemoryMealRepository.java
--- a/src/main/java/ru/javawebinar/topjava/repository/inmemory/InMemoryMealRepository.java	(revision 3b4b9f1d0bbe1b4cac5e6706073ca3c59a532631)
+++ b/src/test/java/ru/javawebinar/topjava/repository/inmemory/InMemoryMealRepository.java	(date 1729116009378)
@@ -20,8 +20,8 @@
 import java.util.function.Predicate;
 import java.util.stream.Collectors;
 
-import static ru.javawebinar.topjava.repository.inmemory.InMemoryUserRepository.ADMIN_ID;
-import static ru.javawebinar.topjava.repository.inmemory.InMemoryUserRepository.USER_ID;
+import static ru.javawebinar.topjava.UserTestData.ADMIN_ID;
+import static ru.javawebinar.topjava.UserTestData.USER_ID;
 
 @Repository
 public class InMemoryMealRepository implements MealRepository {
Index: src/test/java/ru/javawebinar/topjava/web/user/InMemoryAdminRestControllerTest.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/test/java/ru/javawebinar/topjava/web/user/InMemoryAdminRestControllerTest.java b/src/test/java/ru/javawebinar/topjava/web/user/InMemoryAdminRestControllerTest.java
new file mode 100644
--- /dev/null	(date 1729116009305)
+++ b/src/test/java/ru/javawebinar/topjava/web/user/InMemoryAdminRestControllerTest.java	(date 1729116009305)
@@ -0,0 +1,52 @@
+package ru.javawebinar.topjava.web.user;
+
+import org.junit.*;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.context.ConfigurableApplicationContext;
+import org.springframework.context.support.ClassPathXmlApplicationContext;
+import ru.javawebinar.topjava.repository.inmemory.InMemoryUserRepository;
+import ru.javawebinar.topjava.util.exception.NotFoundException;
+
+import java.util.Arrays;
+
+import static ru.javawebinar.topjava.UserTestData.NOT_FOUND;
+import static ru.javawebinar.topjava.UserTestData.USER_ID;
+
+public class InMemoryAdminRestControllerTest {
+    private static final Logger log = LoggerFactory.getLogger(InMemoryAdminRestControllerTest.class);
+
+    private static ConfigurableApplicationContext appCtx;
+    private static AdminRestController controller;
+    private static InMemoryUserRepository repository;
+
+    @BeforeClass
+    public static void beforeClass() {
+        appCtx = new ClassPathXmlApplicationContext("spring/spring-app.xml");
+        log.info("\n{}\n", Arrays.toString(appCtx.getBeanDefinitionNames()));
+        controller = appCtx.getBean(AdminRestController.class);
+        repository = appCtx.getBean(InMemoryUserRepository.class);
+    }
+
+    @AfterClass
+    public static void afterClass() {
+        appCtx.close();
+    }
+
+    @Before
+    public void setup() {
+        // re-initialize
+        repository.init();
+    }
+
+    @Test
+    public void delete() {
+        controller.delete(USER_ID);
+        Assert.assertNull(repository.get(USER_ID));
+    }
+
+    @Test
+    public void deleteNotFound() {
+        Assert.assertThrows(NotFoundException.class, () -> controller.delete(NOT_FOUND));
+    }
+}
\ No newline at end of file
Index: src/main/java/ru/javawebinar/topjava/repository/inmemory/InMemoryUserRepository.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/java/ru/javawebinar/topjava/repository/inmemory/InMemoryUserRepository.java b/src/test/java/ru/javawebinar/topjava/repository/inmemory/InMemoryUserRepository.java
rename from src/main/java/ru/javawebinar/topjava/repository/inmemory/InMemoryUserRepository.java
rename to src/test/java/ru/javawebinar/topjava/repository/inmemory/InMemoryUserRepository.java
--- a/src/main/java/ru/javawebinar/topjava/repository/inmemory/InMemoryUserRepository.java	(revision 3b4b9f1d0bbe1b4cac5e6706073ca3c59a532631)
+++ b/src/test/java/ru/javawebinar/topjava/repository/inmemory/InMemoryUserRepository.java	(date 1729116009421)
@@ -8,11 +8,19 @@
 import java.util.List;
 import java.util.stream.Collectors;
 
+import static ru.javawebinar.topjava.UserTestData.*;
+
+
 @Repository
 public class InMemoryUserRepository extends InMemoryBaseRepository<User> implements UserRepository {
 
-    static final int USER_ID = 1;
-    static final int ADMIN_ID = 2;
+    public void init() {
+        map.clear();
+        put(user);
+        put(admin);
+        put(guest);
+        counter.getAndSet(GUEST_ID + 1);
+    }
 
     @Override
     public List<User> getAll() {
diff --git a/src/main/java/ru/javawebinar/topjava/SpringMain.java b/src/test/java/ru/javawebinar/topjava/SpringMain.java
rename from src/main/java/ru/javawebinar/topjava/SpringMain.java
rename to src/test/java/ru/javawebinar/topjava/SpringMain.java
