Subject: [PATCH] 3_10_add_spring_test
---
Index: pom.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/pom.xml b/pom.xml
--- a/pom.xml	(revision 23bdea5f7c6dcfe985cace4c8e451cfc4eb4b09d)
+++ b/pom.xml	(date 1697664092363)
@@ -103,6 +103,12 @@
             <version>${junit.version}</version>
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <groupId>org.springframework</groupId>
+            <artifactId>spring-test</artifactId>
+            <version>${spring.version}</version>
+            <scope>test</scope>
+        </dependency>
     </dependencies>
 
     <profiles>
Index: src/test/java/ru/javawebinar/topjava/web/user/InMemoryAdminRestControllerSpringTest.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/test/java/ru/javawebinar/topjava/web/user/InMemoryAdminRestControllerSpringTest.java b/src/test/java/ru/javawebinar/topjava/web/user/InMemoryAdminRestControllerSpringTest.java
new file mode 100644
--- /dev/null	(date 1697664100056)
+++ b/src/test/java/ru/javawebinar/topjava/web/user/InMemoryAdminRestControllerSpringTest.java	(date 1697664100056)
@@ -0,0 +1,41 @@
+package ru.javawebinar.topjava.web.user;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit4.SpringRunner;
+import ru.javawebinar.topjava.repository.inmemory.InMemoryUserRepository;
+import ru.javawebinar.topjava.util.exception.NotFoundException;
+
+import static ru.javawebinar.topjava.UserTestData.NOT_FOUND;
+import static ru.javawebinar.topjava.UserTestData.USER_ID;
+
+@ContextConfiguration("classpath:spring/spring-app.xml")
+@RunWith(SpringRunner.class)
+public class InMemoryAdminRestControllerSpringTest {
+
+    @Autowired
+    private AdminRestController controller;
+
+    @Autowired
+    private InMemoryUserRepository repository;
+
+    @Before
+    public void setup() {
+        repository.init();
+    }
+
+    @Test
+    public void delete() {
+        controller.delete(USER_ID);
+        Assert.assertNull(repository.get(USER_ID));
+    }
+
+    @Test
+    public void deleteNotFound() {
+        Assert.assertThrows(NotFoundException.class, () -> controller.delete(NOT_FOUND));
+    }
+}
\ No newline at end of file
